agent/agent.go

```diff
@@ -770,6 +773,7 @@ func (a *Agent) endTaskResponse(tc *taskContext, status string, message string)
 	if tc.taskConfig != nil {
 		detail.Modules.Prefixes = tc.taskConfig.ModulePaths
 	}
+	a.handleTimeoutAndOOM(ctx, tc, status)
```

Comment:
This needs to be moved because it should only run in [cases of success/failure](https://github.com/evergreen-ci/evergreen/blob/0acd94f7c3d69739f39198527bfaa44af1f5e433/agent/agent.go#L678-L690). In the system failed case, timeout cannot run because the task has encountered an unfixable error while gathering necessary task information (such as due to an agent bug, or inability to get information from the app server). Also, it would be a weird combining of responsibilities to run more task logic in a function that's meant to just calculate an end task response. I would move `a.handleTimeoutAndOOM` to call them in the individual success/failed cases.